Jenkinsfile

```diff
@@ -632,6 +638,72 @@ pipeline {
       }
     }
 
+    stage('Deploy Preview') {
+      when {
+        not {
+          branch 'PR-*'
+        }
+      }
+      steps {
+        setGitHubBuildStatus('nuxeo/preview', 'Deploy nuxeo Preview', 'PENDING')
+        container('maven') {
+          dir('ci/helm/preview') {
+            echo """
+            ----------------------------------------
+            Deploy Preview environment
+            ----------------------------------------"""
+            // first substitute environment variables in chart values
+            sh """
+              mv values.yaml values.yaml.tosubst
+              envsubst < values.yaml.tosubst > values.yaml
+            """
+            // second create target namespace (if doesn't exist) and copy secrets to target namespace
+            script {
+              boolean nsExist = sh(returnStatus: true, script: "kubectl get namespace ${PREVIEW_NAMESPACE}") == 0
+              String noCommentOpt = '';
+              if (nsExist) {
+                noCommentOpt = '--no-comment'
+              } else {
+                sh "kubectl create namespace ${PREVIEW_NAMESPACE}"
+              }
+              sh "kubectl create secret generic preview-docker-cfg --namespace=${PREVIEW_NAMESPACE} --from-file=.dockerconfigjson=/home/jenkins/.docker/config.json --type=kubernetes.io/dockerconfigjson --dry-run -o yaml | kubectl apply -f -"
+              // third build and deploy the chart
+              // reference branch have a different preview command. The -n name provided must be lowercase.
+              previewCommand = "jx step helm install --namespace ${PREVIEW_NAMESPACE} -n ${APP_NAME}-${PREVIEW_NAMESPACE.toLowerCase()} ."
+              if (isPullRequest()) {
```

Comment:
Usually, we tend to use either a ternary:
```
def previewCommand = isPullRequest() ?
  "jx preview ..."
  : "jx step helm install..."
```
or the traditional form:
```
if (isPullRequest()) {
  ...
} else {
  ...
}
```
It avoids assigning the variable twice and is easier to read.

Reply:
I went for the ternary to avoid declaring an empty string
```

              // third build and deploy the chart
              previewCommand = isPullRequest()?
                // When deploying a pull request, we use jx preview which gc the merged pull requests every 3h
                // waiting for https://github.com/jenkins-x/jx/issues/5797 to be fixed in order to remove --source-url
                "jx preview --namespace ${PREVIEW_NAMESPACE} --source-url=https://github.com/nuxeo/nuxeo ${noCommentOpt}"
                // To avoid jx gc cron job, reference branch previews are deployed by calling helm install directly.
                //The -n name provided must be lowercase.
                : "jx step helm install --namespace ${PREVIEW_NAMESPACE} -n ${APP_NAME}-${PREVIEW_NAMESPACE.toLowerCase()} ."

```